tests/components/homekit/test_type_humidifiers.py

```diff
@@ -0,0 +1,419 @@
+"""Test different accessory types: HumidifierDehumidifier."""
+from pyhap.const import (
+    CATEGORY_HUMIDIFIER,
+    HAP_REPR_AID,
+    HAP_REPR_CHARS,
+    HAP_REPR_IID,
+    HAP_REPR_VALUE,
+)
+
+from homeassistant.components.homekit.const import (
+    ATTR_VALUE,
+    CONF_LINKED_HUMIDITY_SENSOR,
+    PROP_MAX_VALUE,
+    PROP_MIN_STEP,
+    PROP_MIN_VALUE,
+    PROP_VALID_VALUES,
+)
+from homeassistant.components.homekit.type_humidifiers import HumidifierDehumidifier
+from homeassistant.components.humidifier.const import (
+    ATTR_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    DEFAULT_MAX_HUMIDITY,
+    DEFAULT_MIN_HUMIDITY,
+    DEVICE_CLASS_DEHUMIDIFIER,
+    DEVICE_CLASS_HUMIDIFIER,
+    DOMAIN,
+    SERVICE_SET_HUMIDITY,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_ENTITY_ID,
+    ATTR_UNIT_OF_MEASUREMENT,
+    DEVICE_CLASS_HUMIDITY,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+    UNIT_PERCENTAGE,
+)
+
+from tests.common import async_mock_service
+
+
+async def test_humidifier(hass, hk_driver, events):
+    """Test if humidifier accessory and HA are updated accordingly."""
+    entity_id = "humidifier.test"
+
+    hass.states.async_set(entity_id, STATE_OFF)
+    await hass.async_block_till_done()
+    acc = HumidifierDehumidifier(
+        hass, hk_driver, "HumidifierDehumidifier", entity_id, 1, None
+    )
+    hk_driver.add_accessory(acc)
+
+    await acc.run_handler()
+    await hass.async_block_till_done()
+
+    assert acc.aid == 1
+    assert acc.category == CATEGORY_HUMIDIFIER
+
+    assert acc.char_current_humidifier_dehumidifier.value == 0
+    assert acc.char_target_humidifier_dehumidifier.value == 1
+    assert acc.char_current_humidity.value == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5795327)